### PR TITLE
Including clipper library as conan pkg.

### PR DESCRIPTION
**Pull request by daniellecandid**
Specify library name and version:  **clipper/6.4.2**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/clipper/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.configure(build_folder=self._build_subfolder, source_folder=os.path.join(self._source_subfolder, "cpp"))
+        return cmake
```
Comment (madebr) [inline recipes/clipper/all/conanfile.py:42]: Caching cmake avoid reconfiguration in *package* after *build*.
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.configure(build_folder=self._build_subfolder, source_folder=os.path.join(self._source_subfolder, "cpp"))
        self._cmake
```

File: recipes/clipper/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    settings = "arch", "build_type", "compiler", "os"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+        }
+    default_options = {"shared": False, "fPIC": True}
+
+    _source_subfolder = "source_subfolder"
```
Comment (madebr) [inline recipes/clipper/all/conanfile.py:26]:
```suggestion
    _cmake = None
    _source_subfolder = "source_subfolder"
```

File: recipes/clipper/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        cmake.install()
+
+        self.copy("License.txt", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["polyclipping"]
+        if self.settings.os == "Linux":
+            self.cpp_info.libs.append('pthread')
```
Comment (madebr) [inline recipes/clipper/all/conanfile.py:59]:
```suggestion
            self.cpp_info.system_libs.append("pthread")
```

File: recipes/clipper/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+message(${CONAN_LIBS})
```
Comment (madebr) [inline recipes/clipper/all/test_package/CMakeLists.txt:8]:
```suggestion
```

File: recipes/clipper/config.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+versions:
+    "6.4.2":
+      folder: all
```
Comment (madebr) [inline recipes/clipper/config.yml:3]:
```suggestion
  "6.4.2":
    folder: all
```

File: recipes/clipper/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import os
+from os import path
```
Comment (madebr) [inline recipes/clipper/all/conanfile.py:2]:
```suggestion
```

File: recipes/clipper/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import os
+from os import path
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
```
Comment (madebr) [inline recipes/clipper/all/conanfile.py:5]:
```suggestion
```

**Comment by SpaceIm** [comment]
You should also wrap clipper's CMakeLists.txt with this CMakeLists.txt:

```cmake
cmake_minimum_required(VERSION 2.8.11)
project(cmake_wrapper)

include(conanbuildinfo.cmake)
conan_basic_setup()

add_subdirectory("source_subfolder/cpp")
```

File: recipes/clipper/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/clipper/all/conanfile.py:42]: You can use default `source_folder` in configure if clipper's CMakeLists.txt is properly wrapped by recipe one.

File: recipes/clipper/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        self.copy("License.txt", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["polyclipping"]
```
Comment (SpaceIm) [inline recipes/clipper/all/conanfile.py:57]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "polyclipping"
        self.cpp_info.names["cmake_find_package_multi"] = "polyclipping"
        self.cpp_info.names["pkg_config"] = "polyclipping"
        self.cpp_info.libs = ["polyclipping"]
```

**Comment by SpaceIm** [comment]
I'm also pretty sure that shared will fail with Visual Studio. This library doesn't export symbols. You'll probably have to set `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` to `ON` in CMake wrapper if `WIN32 AND BUILD_SHARED_LIBS` (increasing min CMake version to 3.4).

File: recipes/clipper/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "polyclipping"
+        self.cpp_info.names["cmake_find_package_multi"] = "polyclipping"
+        self.cpp_info.names["pkg_config"] = "polyclipping"
+        self.cpp_info.libs = ["polyclipping"]
+
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("pthread")
```
Comment (SpaceIm) [inline recipes/clipper/all/conanfile.py:64]: Are you sure that this library requires to link `pthread`?

Comment (daniellecandid) [reply recipes/clipper/all/conanfile.py:64]: It definitely doesn't, I'll remove those lines.

**Pull request merged**